src/sinks/util/buffer/compression.rs

```diff
@@ -206,26 +205,69 @@ impl Configurable for Compression {
 
     fn metadata() -> Metadata<Self> {
         let mut metadata = Metadata::default();
-        metadata.set_description("Compression configuration.");
+        metadata.set_title("Compression configuration.");
+        metadata.set_description("All compression algorithms use the default compression level unless otherwise specified.");
         metadata.add_custom_attribute(CustomAttribute::kv("docs::enum_tagging", "external"));
         metadata
     }
 
     fn generate_schema(gen: &mut SchemaGenerator) -> Result<SchemaObject, GenerateError> {
         const ALGORITHM_NAME: &str = "algorithm";
         const LEVEL_NAME: &str = "level";
-        const NONE_NAME: &str = "none";
-        const GZIP_NAME: &str = "gzip";
-        const ZLIB_NAME: &str = "zlib";
-
-        // First, we need to be able to handle all of the string-only variants.
-        let const_values = [NONE_NAME, GZIP_NAME, ZLIB_NAME]
-            .iter()
-            .map(|s| serde_json::Value::from(*s))
-            .collect();
+        const LOGICAL_NAME: &str = "logical_name";
+        const ENUM_TAGGING_MODE: &str = "docs::enum_tagging";
+
+        let generate_string_schema = |logical_name: &str,
+                                      title: Option<&'static str>,
+                                      description: &'static str|
+         -> SchemaObject {
+            let mut const_schema = generate_const_string_schema(logical_name.to_lowercase());
+            let mut const_metadata = Metadata::<()>::with_description(description);
+            if let Some(title) = title {
+                const_metadata.set_title(title);
+            }
+            const_metadata.add_custom_attribute(CustomAttribute::kv(LOGICAL_NAME, logical_name));
+            apply_metadata(&mut const_schema, const_metadata);
+            const_schema
+        };
+
+        // First, we'll create the string-only subschemas for each algorithm, and wrap those up
+        // within a one-of schema.
+        let mut string_metadata = Metadata::<()>::with_description("Compression algorithm.");
+        string_metadata.add_custom_attribute(CustomAttribute::kv(ENUM_TAGGING_MODE, "external"));
+
+        let none_string_subschema = generate_string_schema("None", None, "No compression.");
+        let gzip_string_subschema = generate_string_schema(
+            "Gzip",
+            Some("[Gzip][gzip] compression."),
+            "[gzip]: https://en.wikipedia.org/wiki/Gzip",
```

Comment:
Similar thought of wiki vs "official" links

Reply:
Fixed in 0e79bbe.